### DISCUSSION TITLE
./install.sh don't works on Windows 10 ESP-IDF environment

### DISCUSSION THREAD

**Discussion by mforbiz** [openthread/openthread#9973] [Q&A] [answered]
Hi.
I'm following ESP Tread Border Router installation described here:
[https://openthread.io/guides/border-router/espressif-esp32](url)
I've installed Esp-Idf environment on Windows 10 PC.
Anyway whe I try to insert
`./install.sh`
I receive this error:
_. is not recognized as internal or external command, exe or script_

Any idea?
Thanks in advance

**Comment by jwhui** [comment, marked as answer]
The guide assumes a Linux-based system. If you are using Windows, take a look at [Windows Subsystem for Linux](https://learn.microsoft.com/en-us/windows/wsl/about).

**Reply by mforbiz** [reply]
Hi jwhui.
Thanks for your reply.
Hi don't understand why following the official ESP-IDF Tools Windows Installer guide
[https://docs.espressif.com/projects/esp-idf/en/stable/esp32s3/get-started/windows-setup.html](url)
I need to have a linux environment.
I.e. :
`git clone -b v5.1.2 --recursive https://github.com/espressif/esp-idf.git`
works fine but when I try
`cd esp-idf`
`./install.sh`
I receive the error.
I'm very confused

**Reply by jwhui** [reply]
[install.sh](https://github.com/espressif/esp-idf/blob/master/install.sh) is a bash script.

**Reply by mforbiz** [reply]
Problem solved.
I've installed all on an Ubuntu machine.
Thanks
